aten/src/ATen/core/PhiloxRNGEngine.h

```diff
@@ -192,12 +190,48 @@ class philox_engine {
     ret[3] = lo0;
     return ret;
   }
+
+  C10_HOST_DEVICE constexpr float normalize(uint32_t value) {
+      return value * static_cast<float>(1.0 / std::numeric_limits<uint32_t>::max());
```

Comment:
Due to float rounding, this can produce a value >=1.0.

Reply:
Oh, I expected `==1.0` but not `>1.0`, I suppose something like:

```
float r = static_cast<float>(1.0 / std::numeric_limits<uint32_t>::max())
abs(value * (1-r))
```

Might work? Where r < 1.0 gives us < 1.0, r == 1.0 gives us 0, and r > 1.0 gives us < 1.0 as long as r > 1.0 but < 2.0.

Reply:
https://pytorch.org/docs/stable/generated/torch.rand.html
Need interval `[0,1)` (1-exclusive)

Reply:
You can use the same constant Jason used in Triton to get [0,1), and then use 1-r to get to (0,1]

Reply:
sgtm - ty